x-pack/metricbeat/module/panos/system/resources.go

```diff
@@ -0,0 +1,241 @@
+	5692       20   0  121504   8396   6644 R  94.4   0.2 155491:08 pan_task
+	5695       20   0  121504   7840   6632 R  94.4   0.2 155491:01 pan_task
+	5696       20   0  121504   8360   6816 R  94.4   0.2 155486:29 pan_task
+	5699       20   0  121504   8132   6676 R  94.4   0.2 155236:17 pan_task
+	5700       20   0  146304  18424   6780 R  88.9   0.4 155491:40 pan_task
+
+22360 nobody    20   0  459836  40592  10148 R  22.2   0.8   0:38.65 httpd
+
+	6374       17  -3 1078156  18272   9716 S   5.6   0.4 215:46.41 routed
+
+14227       20   0   18108   7184   2172 R   5.6   0.1   0:00.04 top
+
+	1       20   0    2532    696    656 S   0.0   0.0   3:48.14 init
+	2       20   0       0      0      0 S   0.0   0.0   0:00.83 kthreadd
+*/
+func formatResourceEvents(m *MetricSet, input string) []mb.Event {
+	currentTime := time.Now()
+	events := make([]mb.Event, 0, 1)
+
+	// We only need the top 5 lines
+	lines := strings.Split(input, "\n")
+	lines = lines[:5]
+
+	systemInfo := parseSystemInfo(lines[0])
+	taskInfo := parseTaskInfo(lines[1])
+	cpuInfo := parseCPUInfo(lines[2])
+	memoryInfo := parseMemoryInfo(lines[3])
+	swapInfo := parseSwapInfo(lines[4])
+
+	event := mb.Event{MetricSetFields: mapstr.M{
+		"uptime.days":         systemInfo.Uptime.Days,
+		"uptime.hours":        systemInfo.Uptime.Hours,
+		"user_count":          systemInfo.UserCount,
+		"load_average.1m":     systemInfo.LoadAverage.one_minute,
+		"load_average.5m":     systemInfo.LoadAverage.five_minute,
+		"load_average.15m":    systemInfo.LoadAverage.fifteen_minute,
+		"tasks.total":         taskInfo.Total,
+		"tasks.running":       taskInfo.Running,
+		"tasks.sleeping":      taskInfo.Sleeping,
+		"tasks.stopped":       taskInfo.Stopped,
+		"tasks.zombie":        taskInfo.Zombie,
+		"cpu.user":            cpuInfo.User,
+		"cpu.system":          cpuInfo.System,
+		"cpu.nice":            cpuInfo.Nice,
+		"cpu.idle":            cpuInfo.Idle,
+		"cpu.wait":            cpuInfo.Wait,
+		"cpu.hi":              cpuInfo.Hi,
+		"cpu.system_int":      cpuInfo.SystemInt,
+		"cpu.steal":           cpuInfo.Steal,
+		"memory.total":        memoryInfo.Total,
+		"memory.free":         memoryInfo.Free,
+		"memory.used":         memoryInfo.Used,
+		"memory.buffer_cache": memoryInfo.BufferCache,
+		"swap.total":          swapInfo.Total,
+		"swap.free":           swapInfo.Free,
+		"swap.used":           swapInfo.Used,
+		"swap.available":      swapInfo.Available,
+	},
+		RootFields: mapstr.M{
+			"observer.ip":     m.config.HostIp,
+			"host.ip":         m.config.HostIp,
+			"observer.vendor": "Palo Alto",
+			"observer.type":   "firewall",
+			"@Timestamp":      currentTime,
+		}}
+
+	events = append(events, event)
+	return events
+}
+
+func convertUptime(uptime string) (int, int) {
+	// 07:51
+	hourstr := strings.Split(uptime, ":")
+	hours := parseInt(hourstr[0])
+	minutes := parseInt(hourstr[1])
+
+	return hours, minutes
+}
+
+func parseSystemInfo(line string) SystemInfo {
```

Comment:
Logic in this function is going to break for the example here:

https://www.redhat.com/sysadmin/interpret-top-output#:~:text=BY%2DSA%204.0)-,Uptime,-Top%27s%20first%20line

^^ There are no "days" always.

I think this file needs very careful review. Seems like the simple parser implemented here is not robust; even for other parseXXX functions.